instaloader/structures.py

```diff
@@ -373,14 +373,20 @@ def is_video(self) -> bool:
     @property
     def video_url(self) -> Optional[str]:
         """URL of the video, or None."""
+        def get_size(url):
+            return self._context.head(url, allow_redirects=True).headers.get('Content-Length', 0), url
         if self.is_video:
+            url_candidates = []
+            if 'video_url' in self._node:
+                url_candidates.append(get_size(self._node['video_url']))
             if self._context.is_logged_in:
                 try:
-                    url = self._iphone_struct['video_versions'][0]['url']
-                    return url
+                    for version in self._iphone_struct['video_versions']:
+                        url_candidates.append(get_size(version['url']))
```

Comment:
This would abort the whole candidate selection process if only one of the HEAD requests is answered with a non-ok HTTP response. Maybe it would be better to catch the exceptions inside the loop, to only skip the erroring candidates, and raise an exception only if all candidates yield an error?